src/qibo/transpiler/placer.py

```diff
@@ -0,0 +1,341 @@
+            )
+            if (
+                self.connectivity.number_of_edges()
+                == circuit_subgraph.number_of_edges()
+                or i == len(gates_qubits_pairs) - 1
+            ):
+                break
+        return {"q" + str(i): result.mapping[i] for i in range(len(result.mapping))}
+
+
+class Random(Placer):
+    """
+    Random initialization with greedy policy, let a maximum number of 2-qubit
+    gates can be applied without introducing any SWAP gate.
+
+    Attributes:
+        connectivity (:class:`networkx.Graph`): chip connectivity.
+        samples (int, optional): number of initial random layouts tested.
+            Defaults to :math:`100`.
+    """
+
+    def __init__(self, connectivity, samples: int = 100):
+        self.connectivity = connectivity
+        self.samples = samples
+
+    def __call__(self, circuit):
+        """Find an initial layout of the given circuit using random greedy algorithm.
+
+        Args:
+            circuit (:class:`qibo.models.circuit.Circuit`): Circuit to be transpiled.
+
+        Returns:
+            (dict): physical-to-logical qubit mapping.
+        """
+        gates_qubits_pairs = _find_gates_qubits_pairs(circuit)
+        nodes = self.connectivity.number_of_nodes()
+        keys = list(self.connectivity.nodes())
+        final_mapping = dict(zip(keys, range(nodes)))
+        final_graph = nx.relabel_nodes(self.connectivity, final_mapping)
+        final_mapping = {
+            "q" + str(i): final_mapping[i] for i in range(len(final_mapping))
+        }
+        final_cost = self._cost(final_graph, gates_qubits_pairs)
+        for _ in range(self.samples):
+            mapping = dict(zip(keys, random.sample(range(nodes), nodes)))
+            graph = nx.relabel_nodes(self.connectivity, mapping)
+            cost = self._cost(graph, gates_qubits_pairs)
+            if cost == 0:
+                return {"q" + str(i): mapping[i] for i in range(len(mapping))}
+            if cost < final_cost:
+                final_graph = graph
+                final_mapping = {"q" + str(i): mapping[i] for i in range(len(mapping))}
+                final_cost = cost
+        return final_mapping
+
+    def _cost(self, graph: nx.Graph, gates_qubits_pairs: list):
+        """
+        Compute the cost associated to an initial layout as the lengh of the reduced circuit.
+
+        Args:
+            graph (:class:`networkx.Graph`): current hardware qubit mapping.
+            gates_qubits_pairs (list): circuit representation.
+
+        Returns:
+            (int): lengh of the reduced circuit.
+        """
+        for allowed, gate in enumerate(gates_qubits_pairs):
+            if gate not in graph.edges():
+                return len(gates_qubits_pairs) - allowed - 1
+        return 0
+
+
+class ReverseTraversal(Placer):
+    """
+    Places qubits based on the algorithm proposed in Reference [1].
+
+    Works with ``ShortestPaths`` routing.
```

Comment:
It works with any `Router`, right? 
```suggestion
    Compatible with all the available ``Router``s.
```